x-pack/plugins/fleet/server/services/preconfiguration.ts

```diff
@@ -227,21 +232,25 @@ export async function ensurePreconfiguredPackagesAndPolicies(
       continue;
     }
     fulfilledPolicies.push(policyResult.value);
-    const { created, policy, shouldAddIsManagedFlag } = policyResult.value;
+    const { created, policy, shouldAddIsManagedFlag, namespacedSoClient } = policyResult.value;
+
     if (created || policies[i].is_managed) {
+      if (!namespacedSoClient) {
+        throw new Error('No soClient created for that policy');
```

Comment:
do we expect kibana startup to fail or log an error in some way if the `space_id` is not set up yet?

while testing, I copied your config from the description and kibana started fine and I don't see any fleet errors in the logs. the test policy was created with `"namespace": "default"` when I looked at the raw saved object

Reply:
@jen-huang I should have mentionned it in the test instructions you will have to enable the experimental feature for spaces too.
```
xpack.fleet.enableExperimental: ['useSpaceAwareness']
```

For the error it should never happens, it's just to check a type that could be undefined.